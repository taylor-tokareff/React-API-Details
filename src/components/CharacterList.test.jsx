import React from 'react';
import { render } from '@testing-library/react';
import CharacterList from './CharacterList';
import { MemoryRouter } from 'react-router-dom';

const characters = [...Array(5)].map((_, i) => ({
  character: 'Bender',
  quote: 'Bite my shiny metal ass',
  image: 'url',
})
);

describe('Character component', () => {
  it('renders Character', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CharacterList characters={characters} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
