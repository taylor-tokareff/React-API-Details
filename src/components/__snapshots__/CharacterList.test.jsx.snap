// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <figure>
        <a
          href="/character/Bender"
        >
          <img
            src="url"
          />
        </a>
        <p>
          Bender
        </p>
        <p>
          Bite my shiny metal ass
        </p>
      </figure>
    </li>
    <li>
      <figure>
        <a
          href="/character/Bender"
        >
          <img
            src="url"
          />
        </a>
        <p>
          Bender
        </p>
        <p>
          Bite my shiny metal ass
        </p>
      </figure>
    </li>
    <li>
      <figure>
        <a
          href="/character/Bender"
        >
          <img
            src="url"
          />
        </a>
        <p>
          Bender
        </p>
        <p>
          Bite my shiny metal ass
        </p>
      </figure>
    </li>
    <li>
      <figure>
        <a
          href="/character/Bender"
        >
          <img
            src="url"
          />
        </a>
        <p>
          Bender
        </p>
        <p>
          Bite my shiny metal ass
        </p>
      </figure>
    </li>
    <li>
      <figure>
        <a
          href="/character/Bender"
        >
          <img
            src="url"
          />
        </a>
        <p>
          Bender
        </p>
        <p>
          Bite my shiny metal ass
        </p>
      </figure>
    </li>
  </ul>
</DocumentFragment>
`;
